pkg/signer/file.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2025 The Sigstore Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Copied from https://github.com/sigstore/rekor/blob/c820fcaf3afdc91f0acf6824d55c1ac7df249df1/pkg/signer/file.go
+
+package signer
+
+import (
+	"crypto"
+	"fmt"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"go.step.sm/crypto/pemutil"
+)
+
+// File is a file-based signer/verifier.
+type File struct {
+	signature.SignerVerifier
+}
+
+// NewFileSigner returns an file-based signer and verify, used for spinning up local instances.
+func NewFileSigner(keyPath, keyPass string) (*File, error) {
+	opaqueKey, err := pemutil.Read(keyPath, pemutil.WithPassword([]byte(keyPass)))
+	if err != nil {
+		return nil, fmt.Errorf("file: provide a valid signer, %s is not valid: %w", keyPath, err)
+	}
+
+	signer, err := signature.LoadSignerVerifier(opaqueKey, crypto.SHA256)
```

Comment:
(Also can do in a follow up) Can we change from a hardcoded hash to the typical signature-hash pairing in https://github.com/sigstore/sigstore/pull/2014/files#diff-90f2b413b0fbeae56caac0481e57d7dff1bde02810bce967bd3f8c7d98670010R242?

Any RSA key size uses SHA-256, ECDSA P256/P384/P521 uses SHA-256/384/512 respectively, and Ed25519 uses SHA-512 but that's harcoded into the implementation.

Reply:
I can change this to `signature.LoadDefaultSignerVerifier` which was added ... today.

Reply:
SG, I'll cut a sigstore/sigstore release now

Reply:
https://github.com/sigstore/sigstore/releases/tag/v1.9.1